tools/pyi/gen_pyi.py

```diff
@@ -0,0 +1,249 @@
+from __future__ import print_function
+import os
+import sys
+import inspect
+
+needed_modules = set()
+
+
+def type_to_python(typename):
+    typename = {'int64_t': 'int',
+                'Scalar': 'Union[float, int]',
+                'ScalarType': 'dtype',
+                'Device': 'Union[device, str, None]',
+                'IntList': 'Union[Tuple[int, ...], List[int, ...]]',
+                'Tensor?': 'Optional[Tensor]',
+                'Tensor': 'Tensor',
+                'bool': 'bool',
+                'double': 'float',
+                'Generator *': 'Generator',
+                'Generator*': 'Generator',
+                'std::vector<Tensor>': 'Union[Tuple[Tensor, ...],List[Tensor, ...]]',
+                'TensorList': 'Union[Tuple[Tensor, ...],List[Tensor, ...]]',
+                'Storage': 'Storage',  # inaccurate (FloatStorage...)
+                'SparseTensorRef': 'Tensor',
+                'void': 'None',
+                'Layout': 'layout',
+                'void*': 'int',  # dataptr
+                'std::string': 'str',
+                'real': 'Union[float, int]',
+                'accreal': 'Union[float, int]',
+                'IntegerTensor': 'Tensor',
```

Comment:
nit: I wonder if it would be possible to define some aliases here, so that we can keep a little bit more of the semantic information.